File: Logic Building/2_LOOPING & PATTERNS/4_Pattern Printing/19/19.java
// Q19: Print alphabet pyramid (A, ABA, ABCBA, ABCDCBA).
// Input: An integer n
// Output: Palindrome alphabet pyramid

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                System.out.print(" ");
            }
            for (int j = 0; j <= i; j++) {
                System.out.print((char) ('A' + j));
            }
            for (int j = i - 1; j >= 0; j--) {
                System.out.print((char) ('A' + j));
            }
            System.out.println();
        }
    }
}

File: Logic Building/2_LOOPING & PATTERNS/4_Pattern Printing/19/19.js
// Q19: Print alphabet pyramid (A, ABA, ABCBA, ABCDCBA).
// Input: An integer n
// Output: Palindrome alphabet pyramid

const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin });
rl.on('line', (line) => {
    const n = parseInt(line);
    for (let i = 0; i < n; i++) {
        let row = " ".repeat(n - i - 1);
        for (let j = 0; j <= i; j++) {
            row += String.fromCharCode(65 + j);
        }
        for (let j = i - 1; j >= 0; j--) {
            row += String.fromCharCode(65 + j);
        }
        console.log(row);
    }
    rl.close();
});
